target/linux/generic/backport-5.15/704-15-v5.19-net-mtk_eth_soc-move-MAC_MCR-setting-to-mac_finish.patch

```diff
@@ -25,15 +25,16 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  
  	/* MT76x8 has no hardware settings between for the MAC */
  	if (!MTK_HAS_CAPS(eth->soc->caps, MTK_SOC_MT7628) &&
-@@ -455,16 +455,6 @@ static void mtk_mac_config(struct phylin
+@@ -455,17 +455,6 @@ static void mtk_mac_config(struct phylin
  		return;
  	}
  
 -	/* Setup gmac */
 -	mcr_cur = mtk_r32(mac->hw, MTK_MAC_MCR(mac->id));
 -	mcr_new = mcr_cur;
 -	mcr_new |= MAC_MCR_IPG_CFG | MAC_MCR_FORCE_MODE |
--		   MAC_MCR_BACKOFF_EN | MAC_MCR_BACKPR_EN | MAC_MCR_FORCE_LINK;
+-		   MAC_MCR_BACKOFF_EN | MAC_MCR_BACKPR_EN | MAC_MCR_FORCE_LINK |
+-		   MAC_MCR_RX_FIFO_CLR_DIS;
```

Comment:
You have to add `MAC_MCR_RX_FIFO_CLR_DIS` to the block where it was moved to about 20 lines below this line.

Reply:
I not sure exactly what you're asking me to add... would you mind sending a mergable snippet here as you did last time?

Reply:
This is missing:
````
--- a/target/linux/generic/backport-5.15/704-15-v5.19-net-mtk_eth_soc-move-MAC_MCR-setting-to-mac_finish.patch
+++ b/target/linux/generic/backport-5.15/704-15-v5.19-net-mtk_eth_soc-move-MAC_MCR-setting-to-mac_finish.patch
@@ -43,7 +43,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  	return;
  
  err_phy:
-@@ -478,6 +467,26 @@ init_err:
+@@ -478,6 +467,27 @@ init_err:
  		mac->id, phy_modes(state->interface), err);
  }
  
@@ -58,7 +58,8 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
 +	mcr_cur = mtk_r32(mac->hw, MTK_MAC_MCR(mac->id));
 +	mcr_new = mcr_cur;
 +	mcr_new |= MAC_MCR_IPG_CFG | MAC_MCR_FORCE_MODE |
-+		   MAC_MCR_BACKOFF_EN | MAC_MCR_BACKPR_EN | MAC_MCR_FORCE_LINK;
++		   MAC_MCR_BACKOFF_EN | MAC_MCR_BACKPR_EN | MAC_MCR_FORCE_LINK |
++		   MAC_MCR_RX_FIFO_CLR_DIS;
 +
 +	/* Only update control register when needed! */
 +	if (mcr_new != mcr_cur)
````

Reply:
Thank you for the patch, applied.